oaknational/Oak-Web-Application
# Url helpers + nav menu

## Description

- changes `menuSections` format so that it's more ready to be controlled be CMS
- adds `useIsCurrent` hook for nav links
- fixes menu arrow so that it's still visilbe when there are hashes and/or query params in the url
- adds `LoopingLine` and `LoopingLine2` for nav menus
- updates nav menu colours
- adds `$cover` convenience prop
- adds `OakLink` component
  - makes some preliminary changes to how we handle oak urls
  - groundwork for improvements to how we handle analytics (particularly for cms drivent content)

## Issue(s)

Fixes #446 
Fixes #705 

## How to test

#446

1. Go to {cloud link}
2. Click on menu button
3. You should see the menu like below

#705 

1. Go to https://oak-web-application-q9kd7zzco-oak-national-academy.vercel.app/lesson-planning?23424#abc12
2. Open the menu
3. The arrow should be lined up with "Lesson Plannign"

## Screenshots

![image](https://user-images.githubusercontent.com/12934669/189698367-36457e25-a2e7-4bd7-8852-b09614fef684.png)

![image](https://user-images.githubusercontent.com/12934669/189698425-d96d9d2b-cae8-4fe0-82be-a2f12500b628.png)


## Checklist
